shepherd_bms/src/stateMachine.cpp

```diff
@@ -31,7 +49,12 @@ void StateMachine::handleReady(AccumulatorData_t *bmsdata)
 
 void StateMachine::initCharging()
 {
-    return;
+    if(prevAccData->faultStatus != FAULTS_CLEAR)
+    {
+        requestTransition(FAULTED_STATE);
+    }
+    compute.enableCharging(true);
+    segment.enableBalancing(false);
```

Comment:
i dont wanna run request transmission inside of initialize, because intilize is gonna run inside of req transmission. Not rlly an issue, but gets messy imo. Id rather/ am planning on doing this check in the handle function, and really just want initialize to run the faultcheck function so that we have udpated data entering the handle function

Reply:
I agree, transitioning should happen only in handlers, inits should be solely for setting hardware to the correct state